code/modules/multiz/movement.dm

```diff
@@ -140,14 +157,23 @@
 
 	if(throwing)
 		return
+	if(ishuman(src)) //VOREStation Edit Start. Code to see if they have wings.
+		var/mob/living/carbon/human/H = src
+		if(H.wings_flying == 1) //Some other checks are done in the wings_toggle proc
+			if(H.stat || H.paralysis || H.stunned || H.weakened || H.lying || H.restrained() || H.buckled)
```

Comment:
if(H.incapacitated())

Should work and would be neater.

Reply:
Tried this and it didn't work for whatever reason. Went ahead and did incap check & a few of the checks above. Restrained and buckled were left out since, well, they really aren't needed. Can still flap wings while restrained, and you really shouldn't be able to be buckled while in a space where all objects will fall.

Reply:
In current code you *are* checking for restrained and buckled in `incapacitated()`. Also if you allow flight while buckling, you'll allow flight in wheelchairs, office chairs and similar, which doesn't make too much sense to me. In its current state, it would also get players stuck when the chair falls a level below and they are unable to unbuckle.